lsp.el

```diff
@@ -3290,7 +3292,7 @@ current language. When IGNORE-MULTI-FOLDER is nil current file
 will be openned in multi folder language server if there is
 such."
   (-if-let* ((session (lsp-session))
-             (project-root (lsp--calculate-root session (buffer-file-name)))
+             (project-root (lsp--calculate-root session (or (buffer-file-name) default-directory)))
```

Comment:
Why is `default-directory` used here?

Reply:
When you do find-file and create a new file it won't have buffer-file-name but effectively will be part of the current project.